shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -612,6 +612,7 @@ - (void)flagsChanged:(NSEvent*)event {
 }
 
 - (void)mouseEntered:(NSEvent*)event {
+  _mouseState.has_pending_exit = false;
   [self dispatchMouseEvent:event phase:kAdd];
```

Comment:
You need to gate this on `has_pending_exit`'s status since otherwise you'll send an add without a remove before it.

Reply:
I thought that would redundant since there is already [check for this in dispatchMouseEvent](https://github.com/flutter/engine/blob/master/shell/platform/darwin/macos/framework/Source/FlutterViewController.mm#L435).

Reply:
Ah, I forgot about that. Since that's just a failsafe against weirdness though, I think it would be better to make this explicit here, since it's a deliberate choice in this flow.

Reply:
That makes sense. I'll update the PR tomorrow.